browserdebuggertools/chrome/chromeExtensions/requestBlocker/requestBlocker.js

```diff
@@ -0,0 +1,27 @@
+function blockMainFrames() {
+    chrome.declarativeNetRequest.updateDynamicRules(
+        {
+            addRules: [{
+                "id": 1,
+                "priority": 1,
+                "action": {"type": "block"},
+                "condition": {"resourceTypes": ["main_frame"]}
+            }],
+            removeRuleIds: [1]
```

Comment:
This looks like we are adding and removing the rule again straight away.

Reply:
From https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/

>  The rules with IDs listed in options.removeRuleIds are first removed, and then the rules given in options.addRules are added.

The removeRuleId here was just to make it safe enough to call blockMainFrames more than once consecutively, maybe it's not necessary.